indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -503,6 +513,26 @@ public TaskLockPosse apply(Interval interval)
     }
   }
 
+  public boolean remove(Task task)
+  {
+    giant.lock();
+    try {
+      return activeTasks.remove(task.getId());
```

Comment:
I think this method should be combined with ```unlock(Task)```, they're both called at the same time (by the queue, when a task finishes).

This method should also be signalling the lockReleaseCondition, since when a task is removed then we want to ISE out of any ```lock(task)``` calls for that task.